src/plugins/intel_gpu/src/graph/graph_optimizer/mark_runtime_skippable_nodes.cpp

```diff
@@ -42,6 +43,11 @@ void mark_runtime_skippable_nodes::run(program& p) {
                 GPU_DEBUG_TRACE_DETAIL << "[mark_runtime_skippable_nodes] : " << node.id() << " can_be_optimized" << std::endl;
             }
         });
+        program_helpers::do_for_types<non_max_suppression_gather>(*node, [](non_max_suppression_gather_node& node){
+            node.can_be_optimized(true);
+            node.set_runtime_skippable(true);
+            GPU_DEBUG_TRACE_DETAIL << "[mark_runtime_skippable_nodes] : " << node.id() << " can_be_optimized" << std::endl;
+        });
```

Comment:
I think you can just do this at node creation (like reshape is done) instead of marking here.

Reply:
Updated.